Source/Core/PolylineGeometry.js

```diff
@@ -309,10 +309,45 @@ PolylineGeometry.createGeometry = function (polylineGeometry) {
   var j;
   var k;
 
+  var removedIndices = [];
   var positions = arrayRemoveDuplicates(
     polylineGeometry._positions,
-    Cartesian3.equalsEpsilon
+    Cartesian3.equalsEpsilon,
+    false,
+    removedIndices
   );
+
+  if (defined(colors) && removedIndices.length > 0) {
+    var removedArrayIndex = 0;
+    var length = colors.length;
+    var nextRemovedIndex = removedIndices[0];
+    var colorsScratch = [];
+    var ii;
+    if (colorsPerVertex) {
+      for (ii = 0; ii < length; ii++) {
+        if (ii === nextRemovedIndex) {
+          removedArrayIndex++;
+          nextRemovedIndex = removedIndices[removedArrayIndex];
+        } else {
+          colorsScratch.push(colors[ii]);
+        }
+      }
+    } else {
+      // If polyline is colored in segments, the color is counted
+      // if the endpoint hasn't been removed.
+      for (ii = 0; ii < length; ii++) {
+        if (ii + 1 === nextRemovedIndex) {
+          removedArrayIndex++;
+          nextRemovedIndex = removedIndices[removedArrayIndex];
+        } else {
+          colorsScratch.push(colors[ii]);
+        }
+      }
+    }
+
+    colors = colorsScratch;
+  }
```

Comment:
Is it possible to use [`array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to simplify the code here?